proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,199 @@
+Alternatively, the user can restage the expired package and call the publish command again.
+
+#### Expanding groups
+It should be possible for a group to expand over time. This is currently a real scenario, where .NET GA SDK ships a set of packages,
+ and then follow-up SDK updates ship updated packages which still belong to the same .NET 8 package group.
+ Here how this will play out with the tooling support described above:
+1. During GA release, the release team will stage the set of RTM packages for the release. (nuget stage --group-id "net8.0" --package-id <package-id>)
+2. Then on the day of the release, all the staged packages will be published using `nuget publish --group-id "net8.0"` command.
+   At this point, the stage for the `net8.0` group will be empty.
+3. Later, as new builds are being prepared for a patch release, a new set of packages will be staged, to be later published to the same group.
+As there can be multiple candidate builds for a release, there is a need for functionality to discard a stage for a given group `nuget stage --discard --group-id "net8.0"`.
+This will remove all the staged but not published packages from the net8.0 group.
+4. A new set of packages will be staged for the 8.0 group, untile the final build is known. At this point, there is only one unified set of related packages is staged.
+5. On the release date, the release team will call `nuget publish --group-id "net8.0"` again, and only the staged packages will be published, resulting in an expanded set of published packages in the net8.0 group.
+6. Some time in the future, when the release is already out of support, somebody from the .NET team will call
+ `nuget deprecate --group-id "net8.0"` and all the packages which have ever been published to that group will be deprecated.
+ Those packages, which has already been deprecated because of whatever reason, will not be altered.
+
+#### Asynchronous behavior
+
+As both publishing and deprecation commands can take some time, these will run asynchronously.
+ So the CLI command for both operations will return immediately, giving the user some token which can later be used to
+ track the current state of the request.
+There are no critical scenarios for deprecation to happen within specific time period, however, publishing is somewhat
+ critical and is being orchestrated in many cases. Hence, NuGet should provide some guarantees for how much time the operation
+ can take in the worst case. As of right now, having 10 minutes deadline for the publishing request should be reasonable.
+
+This requirement brings a need for a new NuGet command for querying the status of an asynchronous operation.
+Let's assume that the group publishing and deprecation commands will return some "operation id" for later state inquriry by the client.
+`nuget status --operation-id` command can be used to track the progress of an asynchronous operation.
```

Comment:
Makes sense, but I think this can be put as a stretch goal or secondary priority. This, like listing the contents of a stage, may be easier implemented in UI for now.